internal/controllers/machineset/machineset_controller.go

```diff
@@ -296,26 +305,43 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 		filteredMachines = append(filteredMachines, machine)
 	}
 
+	result := ctrl.Result{}
 	// Remediate failed Machines by deleting them.
 	var errs []error
+	machinesToRemediate := make([]*clusterv1.Machine, 0, len(filteredMachines))
 	for _, machine := range filteredMachines {
-		log := log.WithValues("Machine", klog.KObj(machine))
 		// filteredMachines contains machines in deleting status to calculate correct status.
 		// skip remediation for those in deleting status.
 		if !machine.DeletionTimestamp.IsZero() {
 			continue
 		}
 		if conditions.IsFalse(machine, clusterv1.MachineOwnerRemediatedCondition) {
-			log.Info("Deleting machine because marked as unhealthy by the MachineHealthCheck controller")
-			patch := client.MergeFrom(machine.DeepCopy())
-			if err := r.Client.Delete(ctx, machine); err != nil {
-				errs = append(errs, errors.Wrap(err, "failed to delete"))
-				continue
-			}
-			conditions.MarkTrue(machine, clusterv1.MachineOwnerRemediatedCondition)
-			if err := r.Client.Status().Patch(ctx, machine, patch); err != nil && !apierrors.IsNotFound(err) {
-				errs = append(errs, errors.Wrap(err, "failed to update status"))
+			machinesToRemediate = append(machinesToRemediate, machine)
+		}
+	}
+
+	if len(machinesToRemediate) > 0 {
+		preFlightChecksResult, err := r.runPreFlightChecks(ctx, cluster, machineSet)
```

Comment:
```suggestion
		preflightChecksResult, err := r.runPreflightChecks(ctx, cluster, machineSet)
```
nit: Please check other places as well

Explanation. Either preflight check is one word or two: preflight vs pre-flight. Depending on that I would use camel case or not. Given that we went with preflight elsewhere I would not use preFlight (kubeadm chose the same)